cg/apps/lims/api.py

```diff
@@ -68,6 +64,12 @@ def samples_in_pools(self, pool_name, projectname):
         """Fetch all samples from a pool"""
         return self.get_samples(udf={"pool name": str(pool_name)}, projectname=projectname)
 
+    def get_source(self, lims_id: str) -> str:
```

Comment:
```suggestion
    def get_source(self, lims_id: str) -> str | None:
```

Reply:
I added a note in the docstring too about returning None

Reply:
Sorry I had missed your comment below. I will modify it so the exception is handled in  `get_source` directly and remove it from the validator

Reply:
If there is no sample in LIMS it should return None. Additionally, I don't think we allow `Unknown` anymore from the order portal

Reply:
That was my original thought. However, unknown is not a value that LIMS can register, so in a way, maybe there is no much confusion. 

If @diitaz93 is ok with this behaviour I will revert it again then and document the return of None for whenever there is either no information available or no sample could be fetch

Reply:
discussed and modified